### PR TITLE
Login Page and Servlet

### LINKED ISSUE #124: Create User Authentication Page

### PR DESCRIPTION
**Pull request by anfee1**
Just a simple set up for the authentication and stuff. For some reason I am not able to consistently run the frontend it keeps giving errors and then fixing itself.

Closes: #124

### CODE REVIEW

File: frontend/src/main/java/com/google/tripmeout/frontend/servlet/AuthenticationHandlerServlet.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    response.setContentType("application/json;");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      boolean loggedIn = true;
+      Gson gson = new Gson();
+      String json = gson.toJson(loggedIn);
+      response.getWriter().println(json);
+    } else {
+      boolean loggedIn = false;
```
Comment (cli1903) [inline frontend/src/main/java/com/google/tripmeout/frontend/servlet/AuthenticationHandlerServlet.java]: we're going to need the login/logout link as well, right?

File: frontend/src/main/java/com/google/tripmeout/frontend/AuthenticationInfoModel.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@AutoValue
+public abstract class AuthenticationInfoModel {
+  public static Builder builder() {
+    return new AutoValue_AuthenticationInfoModel.Builder();
+  }
+
+  public abstract String returnLink();
+  public abstract boolean loggedIn();
+  public abstract Builder toBuilder();
+
```
Comment (cli1903) [inline frontend/src/main/java/com/google/tripmeout/frontend/AuthenticationInfoModel.java:14]: suggestion: maybe include user name to use as user id

File: frontend/src/main/java/com/google/tripmeout/frontend/AuthenticationInfoModel.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  @AutoValue.Builder
+  public static abstract class Builder {
+    public abstract String returnLink(String returnLink);
```
Comment (cli1903) [inline frontend/src/main/java/com/google/tripmeout/frontend/AuthenticationInfoModel.java]: I think these should be setter methods that return type Builder

**Review by KoopaKing — APPROVED** [review]

**Review by bsidhom — APPROVED** [review]
LGTM

**Pull request merged**